packages/react-native-reanimated/src/hook/useAnimatedStyle.ts

```diff
@@ -226,7 +239,15 @@ function styleUpdater(
           animationsActive
         );
         if (finished) {
-          last[propName] = updates[propName];
+          if (Array.isArray(updates[propName])) {
+            updates[propName].forEach((obj: StyleProps) => {
+              for (const prop in obj) {
+                last[propName][prop] = obj[prop];
+              }
+            });
+          } else {
```

Comment:
1. When `updates[propName]` is an array?
2. What about spreed operator? 
3. We also need a proper commentary, as this change is not obvious :/
4. Is any change to avoid that copying here? - just wondering 🤔

Reply:
1. transform and boxShadow
2. spread operator removed problem of reference, but only copying prop to prop assured the independence of `toValue` and `current` (in other words it made sure the animation wouldn't jump)
3. on it!
4. in a meaning to avoid copying prop to prop? I tried some other approaches but only this guaranteed smooth animation